### ISSUE TITLE
Include launchSpec in worker-pool error reports

### ISSUE THREAD

**Issue by djmitche** [taskcluster/taskcluster#3164]
https://bugzilla.mozilla.org/show_bug.cgi?id=1650909#c5

> It would be helpful to be able to see the launch-spec associated with a given worker-manager error. I remember seeing in the past errors about an instance type not being available in a given region. I'm not sure if that is what those errors are (but now less helpful) or if it is some other kind of error.

This could probably be as simple as the index of the launchSpec in the worker pool definition.

**Comment by tomprince** [comment]
It would be useful to include the entire spec in the email reports, at least, as those can out-last a given configuration of a worker-pool.

**Comment by djmitche** [comment]
True, but that means carrying it around for every worker, which ends up being a lot of data.  Let's see what it looks like in implementation.

**Comment by tomprince** [comment]
That information would also be necessary for https://github.com/taskcluster/taskcluster-rfcs/pull/161
